File: JS, PHP, PYTHON/PHP/WHILE.php
<?php
function tabla() {
    while (true) {
        $input = readline("Ingrese un número para conocer la tabla de multiplicar: ");
        
        if (is_numeric($input) && intval($input) == $input) {
            $num = (int)$input;
            for ($i = 1; $i <= 10; $i++) {
                echo "$num x $i = " . ($num * $i) . "\n";
            }
            break; // Salir del bucle después de mostrar la tabla
        } else {
            echo "Error, por favor ingrese un número entero\n";
        }
    }
}

tabla();
?>

File: C++ y Java/C++/While.cpp
#include <iostream>
#include <limits>

int main()
{
  while (true)
  {
    try
    {
      std::cout << "Ingrese un numero para conocer la tabla de multiplicar: ";
      int num;
      std::cin >> num;

      if (std::cin.fail())
      {
        throw std::runtime_error("Entrada invalida");
      }

      for (int i = 1; i <= 10; i++)
      {
        std::cout << num << " x " << i << " = " << (num * i) << std::endl;
      }
      break;
    }
    catch (const std::exception &e)
    {
      std::cerr << "Error, por favor ingrese un numero entero" << std::endl;
      std::cin.clear();
      std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    }
  }

  return 0;
}